### PR TITLE
Better QA configuration

### PR DESCRIPTION
**Pull request by FledgeXu**
I'm not sure if we still need to stick `black` and `flake8` to the old version and I updated it for now.

This PR also contains some changes to `.gitignore`, which may not quite fit the principle of one PR for one feature point, but I thought it was a relatively minor change, so I added it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,17 @@ sudo apt install libmagic1 wget ffmpeg \
     libharfbuzz-dev libfribidi-dev libxcb1-dev gifsicle
 ```
 
+# Contribution
+
+```shell
+<Create a Python virtual environments>
```
Comment (rgaudin) [inline README.md:51]: You can remove this line

**Review by rgaudin — CHANGES_REQUESTED** [review]
> I'm not sure if we still need to stick `black` and `flake8` to the old version and I updated it for now.

It's always good to update and keep up with the latest versions. As we were on the previous black version, there's no huge reformat involved but I see one file needing an update. 

```py
black src/ --check
would reformat /Users/reg/src/scraperlib/src/zimscraperlib/image/optimization.py

Oh no! 💥 💔 💥
1 file would be reformatted, 30 files would be left unchanged.
```

You must update tox.ini as well otherwise we'll have discrepencies between the pre-commit hook and the CI. Take a look at the github workflows to understand how it's called.

> This PR also contains some changes to `.gitignore`, which may not quite fit the principle of one PR for one feature point, but I thought it was a relatively minor change, so I added it.

Can you elaborate on why we'd need that? I think it's harmless but for the sake of harmonization, I'd like to know where it's coming from.

Another thing that could be done in this PR is to introduce isort (both in pre-commit and QA). See https://github.com/openzim/wikihow

Of course any non-draft PR should be rebased to main

**Comment by FledgeXu** [comment]
> Can you elaborate on why we'd need that? I think it's harmless but for the sake of harmonization, I'd like to know where it's coming from.

Ok, This is something very annoying that comes with macOS. So When you use the `Finder.app` open any folder in macOS, this application will automatically create a file in that folder called [`.DS_Store`](https://en.wikipedia.org/wiki/.DS_Store).

Maybe I can drop this commit and add it to my global gitignore.

**Comment by rgaudin** [comment]
I know about that one, being on macOS as well. The other ones I understand what they do but is that a list you cooked yourself or is that in the default provided by github or your text editor?

**Comment by rgaudin** [comment]
> @rgaudin It's form the [`gitignore.io`](https://www.toptal.com/developers/gitignore).

Ah that's interesting ; I didn't know this service. What tags did you use? Maybe we could replace it with https://www.toptal.com/developers/gitignore/api/python,macos,windows,linux ? We don't have anything specific in this repo.

**Comment by rgaudin** [comment]
Merging base branch into your feature branch is not rebasing…

**Comment by FledgeXu** [comment]
Sorry, my fault, i push the branch accidentally😅

**Comment by FledgeXu** [comment]
> Ah that's interesting ; I didn't know this service. What tags did you use? Maybe we could replace it with https://www.toptal.com/developers/gitignore/api/python,macos,windows,linux ? We don't have anything specific in this repo.

Yes, we can. It's a pretty handy website.

**Comment by rgaudin** [comment]
> Yes, we can.

Please do ; it's a good opportunity.

**Comment by FledgeXu** [comment]
This PR is ready for review.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -52,3 +52,9 @@ jobs:
       run: tox -e py38,black,coverage
       env:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+    - name: Check import order with isort
```
Comment (rgaudin) [inline .github/workflows/ci.yml:56]: You didn't delete it from here…

File: tox.ini [hunk 3]
```diff
@@ -15,9 +15,14 @@ commands =
...
 deps =
-    black>=22.3.0,<23
+    black>=23.1.0,<24
 commands = black --check .
 
+[testenv:isort]
+deps =
+    isort>=5.12.0,<5.13
+commands = isort --profile black --check src tests
+
```
Comment (rgaudin) [inline tox.ini:25]: While we're at it ; can we add flake8 as well ?

```ini
[testenv:flake8]
deps =
    flake8>=6.0.0,<7.0
commands = flake8 src --count --max-line-length=88 --statistics --extend-ignore=E203
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -52,3 +52,9 @@ jobs:
       run: tox -e py38,black,coverage
```
Comment (rgaudin) [inline .github/workflows/ci.yml:52]: Must now call isort (and flake8)

```yaml
  run: tox -e py38,black,isort,flake8,coverage
```

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thanks but you forgot to remove it from the workflow and it is not called neither…
I've used the opportunity to suggest adding flake8 as well

File: tox.ini (hunk 3 above)
Comment (FledgeXu) [reply tox.ini:25]: Do We need use `flake8` to check tests?

Comment (rgaudin) [reply tox.ini:25]: It's not testing the tests, it's testing `src/`

**Pull request merged**